OBSDOCS-1806: Tailored Network Policies for Loki Operator (#100718)
openshift/openshift-docs: modules/configuring-a-network-policy-for-loki.adoc

:_newdoc-version: 2.18.4
:_template-generated: 2025-10-17
:_mod-docs-content-type: PROCEDURE

[id="configuring-a-network-policy-for-loki_{context}"]
= Configuring a network policy for Loki

> You can configure Loki network policy with the `networkPolicies` field in your LokiStack specification.

Reviewer: ```suggestion
You enable or disable the deployment of `NetowrkPolicies` per LokiStack by setting the field `networkPolicies`. 
```

> Enable or disable the deployment of `NetowrkPolicies` per LokiStack by setting the `networkPolicies` field. 

Author: good catch, fixed

.Prerequisites
* You have administrator permissions.
* You have installed the {oc-first}.
* You have installed the {loki-op}.
* You have created a `LokiStack` custom resource (CR).

.Procedure
. Update the `LokiStack` CR:
+
[source,yaml]
----
apiVersion: loki.grafana.com/v1
kind: LokiStack
metadata:
  name: logging-loki
  namespace: openshift-logging
spec:
  size: 1x.small
  storage:
    schemas:
    - version: v13
      effectiveDate: "<yyyy>-<mm>-<dd>"
    secret:
      name: logging-loki-s3
      type: s3
  storageClassName: <storage_class_name>
  tenants:
    mode: openshift-logging
  networkPolicies:
    disabled: false  # Enable network policies
----
+

> You can set one of the following values for the `networkPolicies.disabled` field:

Reviewer: ```suggestion
You can set one of the following values for the `spec.networkPolicies.disabled` field:
```

+
`None`::

> {loki-op} will not create any network policy.

Reviewer 2: suggest to make it to 'will not deploy' instead of create.

Author: done

> No network policy is created and full network access is allowed to Loki resources.

Reviewer: ```suggestion
{loki-op} will delete any network policy that was previously managing
```

`RestrictIngressEgress`::

> {loki-op} will deploy a set of network policies that restrict the the communications to and from the Loki components.

Author: fixed

> Loki network policies are enforced. Network access to Loki resources is restricted with explicit allow rules.

Reviewer: ```suggestion
{loki-op} will deploy a set of network policies that restrict the the communications to and from the Loki components.```
